packages/react-components/react-message-bar/library/src/components/MessageBar/useMessageBarReflow.ts

```diff
@@ -27,9 +27,20 @@ export function useMessageBarReflow(enabled: boolean = false) {
       }
 
       const entry = entries[0];
+
+      if (!entry) {
+        return;
+      }
+
       // `borderBoxSize` is not supported before Chrome 84, Firefox 92, nor Safari 15.4
-      const inlineSize = entry?.borderBoxSize?.[0].inlineSize ?? entry?.target.getBoundingClientRect().width;
-      if (inlineSize === undefined || !entry) {
+      let inlineSize: number | undefined;
+      if (entry.borderBoxSize && entry.borderBoxSize.length > 0) {
```

Comment:
Nit for conciseness
```suggestion
      if (entry.borderBoxSize?.length) {
```

Reply:
Actually, I'm not sure what this if-else block is meant to do not already done by the code replacing it. They're functionally equivalent.

Reply:
The difference is that the previous code nullish checked `borderBoxSize` and the new code does that and also checks that `borderBoxSize` has elements.

Until this issue came up I was unaware `borderBoxSize` [might be empty](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry/borderBoxSize#examples) in which case accessing `borderBoxSize?.[0].inlineSize` would throw the observed error.

I like your conciseness change!

Reply:
@spmonahan No, if an optional chaining operator `?.` is used on a property or array reference (as in `borderBoxSize?.[0]`), then if the object of the lookup is `null` or `undefined`, then the rest of the lookup is skipped.

```ts
if (borderBoxSize === undefined) {
  assert(borderBoxSize?.[0].inlineSize === undefined); // No Type error
}
```

You can try this yourself with `node -i` or the browser dev tools.

Reply:
@MLoughry The issue arises when `borderBoxSize` is defined but an empty array:

```js
let borderBoxSize = [];
assert(borderBoxSize?.[0].inlineSize === undefined); // <-- Type error
```

To handle the empty array case we either need my existing change or:

```js
let borderBoxSize = [];
assert(borderBoxSize?.[0]?.inlineSize === undefined); // No Type error
```

<img width="425" alt="msedge_pVH6NHsxa3" src="https://github.com/user-attachments/assets/8c446b0e-955d-488f-9887-84f8db8782d6">


```